### PR TITLE
Fix sorting of import completion items

### PR DESCRIPTION
**Pull request by genlu**
Make sure type with shorter name shows first, e,g, `SomeType` before `SomeTypeWithLongerName`. Current implementation would show "SomeTypeWithLongerName" first.

Thanks @Dreamescaper for dogfooding and reporting the issue :)

@dotnet/roslyn-ide @CyrusNajmabadi

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/TypeImportCompletionProviderTests.cs [hunk 1]
```diff
@@ -646,6 +646,43 @@ class Bat
             await VerifyTypeImportItemIsAbsentAsync(markup, "Barr", inlineDescription: "Foo.Bar");
         }
 
+        [InlineData(true)]
+        [InlineData(false)]
+        [Theory, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TypesWithIdenticalNameButDifferentNamespaces(bool isProjectReference)
```
Comment (genlu) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/TypeImportCompletionProviderTests.cs]: This test is unrelated to this fix, just to plug a test gap.

File: src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs [hunk 2]
```diff
@@ -29,7 +29,8 @@ public static CompletionItem Create(INamedTypeSymbol typeSymbol, string containi
 
             // Hack: add tildes (ASCII: 126) to name and namespace as sort text:
             // 1. '~' before type name makes import items show after in-scope items
-            // 2. '~' before namespace makes types with identical type name but from different namespace all show up in the list
+            // 2. ' ' before namespace makes types with identical type name but from different namespace all show up in the list,
+            //    it also makes sure type with shorter name shows first, e.g. 'SomeType` before 'SomeTypeWithLongerName'.
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs]: Do we not have any tests that assert the relative ordering of two items? Don't have the test directly assert the sort text of course, but I'd expect we'd recreate this and at least say "this item appears before that item"?

Comment (genlu) [reply src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs]: I will add a test for this

Comment (dpoeschl) [reply src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs]: We have `AssertItemsInOrder` but you have to give it the whole list. Maybe that thing could take a flag?

Comment (genlu) [reply src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs]: Test added

**Comment by genlu** [comment]
@jinujoseph @vatsalyaagrawal For approval
